OSDOCS-16850 2 - CQA for OVNK-1 (#111189)
openshift/openshift-docs: modules/nw-ovn-kubernetes-list-database-contents.adoc

[id="nw-ovn-kubernetes-list-database-contents_{context}"]
= Listing the OVN-Kubernetes northbound database contents

[role="_abstract"]

> Each node is controlled by the `ovnkube-controller` container running in the `ovnkube-node` pod on that node. To understand the OVN logical networking entities you need to examine the northbound database that is running as a container inside the `ovnkube-node` pod on that node to see what objects are in the node you want to see.

Reviewer: ```suggestion
To understand the OVN-Kubernetes (OVN-K) logical networking entities, you must examine the northbound database that is running as a container inside the `ovnkube-node` pod on that node to see what objects are in the node.

Each node is controlled by the `ovnkube-controller` container running in the `ovnkube-node` pod on that node. 
```
ugh this ole stuff! maybe something like that?

Author: [role="_abstract"]
To understand OVN-Kubernetes (OVN-K) logical networking entities, you must examine the northbound database running as a container inside the `ovnkube-node` pod. Each node is controlled by the `ovnkube-controller` container running within that local pod structure.

I'm thinking this addresses your concerns and Joe's comments below...

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

> * The OpenShift CLI (`oc`) installed.

Reviewer: ```suggestion
* The {oc-first) is installed.
```

Author: ty

> * To run `ovn-nbctl` or `ovn-sbctl` commands in a cluster, open a remote shell into the `nbdb` or `sbdb` containers on the relevant node.

Reviewer: this should be the first step of the procedure b/c the rest probably won't work without it; i would just put it there and make a sep bug jira to get a command added later

Author: I kept list-pods first because oc rsh needs a pod name (ovnkube-node-55xs2). This has been moved to step 3 without a command.
 https://redhat.atlassian.net/browse/OCPBUGS-88579 is the JIRA I made to get a command added later- I'll fill in details after merge.

.Procedure

. List pods by running the following command:
+
[source,terminal]
----
$ oc get po -n openshift-ovn-kubernetes
----
+

> *Example output*

Reviewer: ```suggestion
.Example output
```
revert or choose a listed solution (see the `BlockTitle` on the confluence page if you feel you have to do something with these; tiger team guidance on these simple ones is to leave them as-is)

Author: Thanks, missed the guidance to leave these; it got flagged by vale. Figured it was an easy fix to bold it now as opposed to leaving it for later.

$ oc get pods -n openshift-ovn-kubernetes -owide
----
+

> *Example output*

Reviewer: revert

-c northd -- ovn-nbctl lr-list
----
+

> *Example output*

Reviewer: revert

-c nbdb -- ovn-nbctl ls-list
----
+

> *Example output*

Reviewer: revert

-c nbdb -- ovn-nbctl lb-list
----
+

> *Example output*

Reviewer: revert
